### DISCUSSION TITLE
tweaks "hide my own posts" used to hide what i reblogged

### DISCUSSION THREAD

**Discussion by corvanablack** [AprilSylph/XKit-Rewritten#586] [Suggestion Box]
i don't know if it's still possible for it to work this way, but i used the "hide my own posts" feature to hide the posts i'd reblogged and it would hide every version of it for a very long while. since i've got memory issues and there's no way for me to know if i've reblogged or queued something already without going to check first, i really enjoyed that feature! so i was wondering if it was possible for there to be a toggle or... something i don't know, to make it work like that again? or something else to do that?

if it's not something that's possible anymore that's alright as well, but i thought i'd ask just in case!

**Comment by marcustyphoon** [comment]
Oh, that's interesting - I'll have to check, but I don't remember "hide my own posts" ever working like that!

**edit:** Oh, I see, you're probably thinking of the `Hide posts if they are AlreadyReblogged on my dashboard` option in XKit 7, I think?

This is a somewhat difficult thing to implement, in general, because you might have reblogged upwards of tens of thousands of posts, and checking any particular post against every single one of those would require XKit to have the entire post history of your blog downloaded.

Quick Reblog makes the reblog button green when it knows that you're reblogged a post before, which sounds like just what you want! But note that it only knows that you've reblogged something if you use the XKit Rewritten Quick Reblog popup to do it.

<img width="385" src="https://user-images.githubusercontent.com/8336245/167029088-73801c1f-a611-424c-a71f-4bda80b27972.png">


(Perhaps we could implement a way to populate this list with the last 10000 reblogs on your main blog, but that's not so simple, either. Right now it isn't blog-specific!)

**Reply by corvanablack** [reply]
i can say it definitely wasn't EVERY post, as i'd see ones that i knew i reblogged (some of those "forever reblog" types) pop up over time, but it was probably... the last 1-2 thousand? enough of a gap that i wasn't bothered with reblogging again! but yes, seeing your edit that might have been it? like i mentioned before, my memory is terrible, so i'm just going off of what i can see right now XD

the green reblog helps, and i didn't know that i could change it to 10,000 posts! perhaps that will be enough for me? it was at 100 posts and that's just... nowhere near enough XD i do prefer not seeing them at all, but if it's not feasible to do that, then the 10k posts thing should cover me!

thanks so much for replying!
